lib/core/src/Cardano/Wallet/Primitive/BlockSummary.hs

```diff
@@ -184,7 +192,8 @@ nullBlockEvents BlockEvents{transactions,delegations}
 mergeSublist :: Sublist a -> Sublist a -> Sublist a
 mergeSublist (All xs) _ = All xs -- result cannot be larger
 mergeSublist _ (All ys) = All ys
-mergeSublist (Some xs) (Some ys) = Some $ mergeOn fst const xs ys
+mergeSublist (Some xs) (Some ys) =
+    unsafeMkSublist $ mergeOn fst const (Map.toList xs) (Map.toList ys)
```

Comment:
```suggestion
mergeSublist (Some xs) (Some ys) = Some $ Map.union xs ys
```

This is a benefit of using `Map` 😊 — I should have implemented like this before. I believe that `mergeOn` can actually be removed now.